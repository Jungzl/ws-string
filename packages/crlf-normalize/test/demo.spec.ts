import { crlf, LF, CRLF, CR, chkcrlf } from '../index';

const text = 'foo\r\nbar\nbaz\r';

test(`LF`, () =>
{
	let actual = crlf(text, LF);

	expect(actual).not.toMatch(/\r/)
	expect(actual).toMatchSnapshot();
});

test(`CRLF`, () =>
{
	let actual = crlf(text, CRLF);

	expect(actual.replace(/\r\n/g, '')).toStrictEqual(actual.replace(/\s/g, ''))
	expect(actual).toMatchSnapshot();
});


test(`CR`, () =>
{
	let actual = crlf(text, CR);

	expect(actual).not.toMatch(/\n/)
	expect(actual).toMatchSnapshot();
});

test(`chkcrlf`, () =>
{
	let actual = chkcrlf(text);

	expect(actual).toMatchSnapshot();
});
