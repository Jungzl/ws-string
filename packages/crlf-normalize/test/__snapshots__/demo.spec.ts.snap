// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CR 1`] = `
"foo
bar
baz
"
`;

exports[`CRLF 1`] = `
"foo
bar
baz
"
`;

exports[`LF 1`] = `
"foo
bar
baz
"
`;

exports[`chkcrlf 1`] = `
{
  "cr": true,
  "crlf": true,
  "lf": true,
}
`;

exports[`detectLineBreak 1`] = `
"
"
`;
